backend/algorithms/objects/user.py

```diff
@@ -153,7 +153,7 @@ def wam(self, category: Category = AnyCategory()) -> Tuple[Optional[float], bool
                     counted_uoc += uoc
                     total_mark += uoc * grade
 
-        return (None if counted_uoc == 0 else total_mark / counted_uoc, total_uoc == counted_uoc)
+        return (None if counted_uoc == 0 else total_mark / counted_uoc, total_uoc == counted_uoc and counted_uoc != 0)
```

Comment:
technically, if counted_uoc == 0, then all uoc do have an associated mark?

Reply:
I did it like this because from a user perspective, we can't tell if a user just forgot to put in their marks or they actually did not have a mark. Same assumption as UNSW. they assume you have an error wam till the end of your first term

Reply:
Wait so (writing out so i fully understand), currently we use this boolean to tell whether we say a course is unlocked or not (ie the warnings are "soft" or "hard" warnings). 

So the idea behind it was that, if we are certain you do not meet the wam requirement (since you have entered all marks for every course you have), then we will say the course is locked (and the warnings will be hard warnings). Otherwise, if you have not entered all your marks, then we are not certain a WAM requirement is accurately met, thus we still let the course be unlocked. 

Currently, if
- no marks and no courses, then we are certain you do not make WAM requirement (since you have no WAM)
- 0 to some marks and some courses, we cannot be certain
- all marks for all courses, we can be certain on warnings

So by making this change, you are suggesting that if they have no courses and no marks, then we cannot be certain? (case 1)

Reply:
i actually agree

Reply:
Yea